{"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/dxfeedframework\/promise\/completeexceptionally(_:)"]}],"primaryContentSections":[{"kind":"declarations","declarations":[{"languages":["swift"],"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"completeExceptionally","kind":"identifier"},{"kind":"text","text":"("},{"text":"_","kind":"externalParam"},{"text":" ","kind":"text"},{"text":"exception","kind":"internalParam"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/GraalException"},{"text":") ","kind":"text"},{"text":"throws","kind":"keyword"}],"platforms":["macOS"]}]},{"content":[{"level":2,"text":"Discussion","type":"heading","anchor":"discussion"},{"type":"unorderedList","items":[{"content":[{"inlineContent":[{"text":"Throws : GraalException. Rethrows exception from Java","type":"text"}],"type":"paragraph"}]}]}],"kind":"content"}],"abstract":[{"type":"text","text":"Completes computation exceptionally with a specified exception."},{"type":"text","text":" "},{"text":"This method does nothing if computation has already ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()"},{"type":"text","text":" completed,"},{"text":" ","type":"text"},{"text":"otherwise ","type":"text"},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","isActive":true},{"type":"text","text":" will return the specified exception."}],"metadata":{"symbolKind":"method","role":"symbol","title":"completeExceptionally(_:)","externalID":"s:15DXFeedFramework7PromiseC21completeExceptionallyyyAA14GraalExceptionOKF","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"completeExceptionally"},{"kind":"text","text":"("},{"text":"GraalException","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","kind":"typeIdentifier"},{"kind":"text","text":") "},{"kind":"keyword","text":"throws"}],"roleHeading":"Instance Method","modules":[{"name":"DXFeedFramework"}]},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise"]]},"schemaVersion":{"patch":0,"major":0,"minor":3},"sections":[],"identifier":{"url":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","interfaceLanguage":"swift"},"references":{"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework":{"url":"\/documentation\/dxfeedframework","kind":"symbol","role":"collection","abstract":[],"title":"DXFeedFramework","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","type":"topic"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/cancel()":{"kind":"symbol","url":"\/documentation\/dxfeedframework\/promise\/cancel()","role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"cancel","kind":"identifier"},{"kind":"text","text":"()"}],"title":"cancel()","type":"topic","abstract":[{"text":"Cancels computation. This method does nothing if computation has already ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()"},{"text":" completed.","type":"text"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/complete(result:)":{"url":"\/documentation\/dxfeedframework\/promise\/complete(result:)","kind":"symbol","role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"complete","kind":"identifier"},{"text":"(","kind":"text"},{"text":"result","kind":"externalParam"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework11MarketEventC","text":"MarketEvent"},{"kind":"text","text":") "},{"text":"throws","kind":"keyword"}],"title":"complete(result:)","type":"topic","abstract":[{"text":"Completes computation normally with a specified result.","type":"text"},{"text":" ","type":"text"},{"text":"This method does nothing if computation has already ","type":"text"},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","isActive":true},{"type":"text","text":" completed"},{"type":"text","text":" "},{"type":"text","text":"(normally, exceptionally, or was cancelled),"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isDone()":{"kind":"symbol","url":"\/documentation\/dxfeedframework\/promise\/isdone()","role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"isDone"},{"kind":"text","text":"() -> "},{"text":"Bool","kind":"typeIdentifier","preciseIdentifier":"s:Sb"}],"title":"isDone()","type":"topic","abstract":[{"text":"Returns ","type":"text"},{"type":"strong","inlineContent":[{"type":"text","text":"true"}]},{"type":"text","text":" when computation has"},{"text":" ","type":"text"},{"isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","type":"reference"},{"type":"text","text":" completed normally,"},{"type":"text","text":" "},{"type":"text","text":"or "},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","isActive":true},{"type":"text","text":" exceptionally,"},{"type":"text","text":" "},{"text":"or was ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"text":" canceled","type":"text"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/completeExceptionally(_:)":{"url":"\/documentation\/dxfeedframework\/promise\/completeexceptionally(_:)","kind":"symbol","role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"completeExceptionally","kind":"identifier"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException","kind":"typeIdentifier"},{"text":") ","kind":"text"},{"kind":"keyword","text":"throws"}],"title":"completeExceptionally(_:)","type":"topic","abstract":[{"type":"text","text":"Completes computation exceptionally with a specified exception."},{"type":"text","text":" "},{"type":"text","text":"This method does nothing if computation has already "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()"},{"text":" completed,","type":"text"},{"type":"text","text":" "},{"text":"otherwise ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()"},{"text":" will return the specified exception.","type":"text"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise":{"kind":"symbol","url":"\/documentation\/dxfeedframework\/promise","navigatorTitle":[{"text":"Promise","kind":"identifier"}],"role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise","fragments":[{"text":"class","kind":"keyword"},{"text":" ","kind":"text"},{"text":"Promise","kind":"identifier"}],"title":"Promise","type":"topic","abstract":[{"type":"text","text":"Result of a computation that will be completed normally or exceptionally in the future."},{"type":"text","text":" "},{"type":"text","text":"This class is designed to represent a promise to deliver certain result."}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isCancelled()":{"type":"topic","role":"symbol","abstract":[{"type":"text","text":"Returns "},{"inlineContent":[{"type":"text","text":"true"}],"type":"strong"},{"type":"text","text":" when computation was canceled."},{"type":"text","text":" "},{"type":"text","text":"Use "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","isActive":true,"type":"reference"},{"type":"text","text":" method to get the corresponding exception."}],"title":"isCancelled()","url":"\/documentation\/dxfeedframework\/promise\/iscancelled()","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"isCancelled"},{"text":"() -> ","kind":"text"},{"text":"Bool","kind":"typeIdentifier","preciseIdentifier":"s:Sb"}],"kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isCancelled()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/hasException()":{"type":"topic","role":"symbol","abstract":[{"type":"text","text":"Returns "},{"inlineContent":[{"text":"true","type":"text"}],"type":"strong"},{"type":"text","text":" when computation has completed exceptionally or was cancelled."},{"type":"text","text":" "},{"type":"text","text":"Use "},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","isActive":true,"type":"reference"},{"type":"text","text":" method to get the exceptional outcome of the computation."}],"title":"hasException()","url":"\/documentation\/dxfeedframework\/promise\/hasexception()","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"hasException"},{"kind":"text","text":"() -> "},{"preciseIdentifier":"s:Sb","kind":"typeIdentifier","text":"Bool"}],"kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/hasException()"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/GraalException":{"navigatorTitle":[{"text":"GraalException","kind":"identifier"}],"title":"GraalException","url":"\/documentation\/dxfeedframework\/graalexception","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/GraalException","role":"symbol","type":"topic","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"GraalException","kind":"identifier"}],"abstract":[{"type":"text","text":"Represents errors that occur when calling graal function."}],"kind":"symbol"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/getException()":{"url":"\/documentation\/dxfeedframework\/promise\/getexception()","kind":"symbol","role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/getException()","fragments":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"getException"},{"kind":"text","text":"() -> "},{"kind":"typeIdentifier","preciseIdentifier":"s:15DXFeedFramework14GraalExceptionO","text":"GraalException"},{"kind":"text","text":"?"}],"title":"getException()","type":"topic","abstract":[{"text":"Returns exceptional outcome of computation. If computation has no  ","type":"text"},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/hasException()"},{"text":" exception,","type":"text"},{"type":"text","text":" "},{"text":"then this method returns nil. If computation has completed exceptionally or was cancelled, then","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"the result of this method is not nil."},{"type":"text","text":" "},{"type":"text","text":"If computation was "},{"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isCancelled()","isActive":true},{"type":"text","text":" cancelled, then this method returns an"},{"text":" ","type":"text"},{"text":"instance of GraalException.","type":"text"}]}}}